File: backend/internal/tables/audit.go
package tables

import (
	"time"
)

type Audit struct {
	ID           uint      `json:"id"`
	CreatedAt    time.Time `json:"createdAt"`
	UserId       uint      `json:"userId"`
	Action       string    `json:"action"`
	ActionParams string    `gorm:"type:json" json:"actionParams"`
	IpAddress    string    `json:"ipAddress"`
}

File: frontend/src/interfaces/audit.ts
import {Type} from "class-transformer";

export class Audit {
  id!: number;
  createdAt!: string;
  userId!: number;
  action!: string;
  actionParams!: string;
  ipAddress!: string;
}

export class Pagination {
  page!: number;
  size!: number;
  total!: number;
}

export class AuditListResponse {
  @Type(() => Audit)
  results!: Audit[];
  @Type(() => Pagination)
  pagination!: Pagination;
}
